lib/sensu/api/process.rb

```diff
@@ -334,7 +334,17 @@ def resolve_event(event_json)
           unless clients.empty?
             clients.each_with_index do |client_name, index|
               settings.redis.get("client:#{client_name}") do |client_json|
-                response << MultiJson.load(client_json)
+                begin
+                  response << MultiJson.load(client_json)
+                rescue MultiJson::ParseError
+                  settings.logger.error(
+                    "issue loading json for client #{client_name}: " \
+                    "redis returned #{client_json.inspect}")
+                  if client_json.nil?
+                    settings.redis.srem("clients", client_name)
+                  end
+                  next
```

Comment:
Wouldn't `next` break the async response in the event of an error in the last client?